Source/WebCore/platform/graphics/egl/GLContext.cpp

```diff
@@ -381,6 +381,12 @@ GLContext::GLContext(PlatformDisplay& display, EGLContext context, EGLSurface su
     ASSERT(type == Surfaceless || surface != EGL_NO_SURFACE);
     RELEASE_ASSERT(m_display.eglDisplay() != EGL_NO_DISPLAY);
     RELEASE_ASSERT(context != EGL_NO_CONTEXT);
+
+    const char* extensions = eglQueryString(display.eglDisplay(), EGL_EXTENSIONS);
+    if (GLContext::isExtensionSupported(extensions, "EGL_KHR_swap_buffers_with_damage"))
+        m_eglSwapBuffersWithDamage = eglSwapBuffersWithDamageKHR;
+    else if (GLContext::isExtensionSupported(extensions, "EGL_EXT_swap_buffers_with_damage"))
+        m_eglSwapBuffersWithDamage = eglSwapBuffersWithDamageEXT;
```

Comment:
These extensions don't depend on the context, but on the display, so we could handle them in PlatformDisplay like we do for other extensions. I don't think we need set the function as a member of the instance, the heavy part is checking the extension, which will be done once by PlatformDisplay, here we can just check the extension and use one or the other when needed.

Reply:
done